### PR TITLE
[service.upnext] v1.0.0

### PR DESCRIPTION
**Pull request by im85288**
### Description
service.upnext is a replacement for service.nextup.notification. 
### Checklist:

- [ x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [ x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [ x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: service.upnext/addon.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <extension point="xbmc.addon.metadata">
+        <summary lang="en">Provides a netflix style pop up notification for watching the next episode with fancy/simple
+            modes.
+        </summary>
+        <description lang="en">Provides a netflix style pop up notification for watching the next episode with
+            fancy/simple modes. A prompt to check if your still watching also occurs after a few automatically watched
+            shows in a row.
+        </description>
+        <platform>all</platform>
+        <language/>
```
[the remark this answers is not in the document]
Comment (im85288) [reply service.upnext/addon.xml]: I toned down the logging to what I feel is an acceptable level?

When I removed the language tags, the travis build failed. Are you saying it should be something else with the en_GB instead?

**Comment by im85288** [comment]
Btw I’m using IntelliJ, do you know how to squash commits with that. Tried interactive rebase last time and although it looked good to me it wasn’t one commit.

**Comment by Razzeee** [comment]
Did you force push after that?

**Comment by Rechi** [comment]
I don't know any GUI tools that can squash or amend commits, use the command line.

In your previous PR it looked like you missed `-f` in your git push command (`git push -f origin leia` for pushing rewritten commits, which you already have pushed before).

**Comment by im85288** [comment]
Hmm well I’ll try again with the language change if that’s still needed?

File: service.upnext/addon.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rechi) [reply service.upnext/addon.xml]: If you remove this tag (`<language/>`) addon-checker shouldn't report an error.

File: service.upnext/addon.xml [hunk 2]
```diff
@@ -5,15 +5,14 @@
     </requires>
     <extension library="service.py" point="xbmc.service"/>
     <extension point="xbmc.addon.metadata">
-        <summary lang="en_GB">Provides a netflix style pop up notification for watching the next episode with fancy/simple
```
Comment (Rechi) [inline service.upnext/addon.xml]: I didn't tell to remove the lang attributes.

Comment (im85288) [reply service.upnext/addon.xml]: Ahh ok I'll put them back

Comment (enen92) [reply service.upnext/addon.xml]: I asked for the removal of the language element from metadata not the attributes for summary and description

**Comment by im85288** [comment]
Any movement here guys?

File: service.upnext/addon.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version='1.0' encoding='UTF-8' standalone='yes'?>
+<addon id="service.upnext" name="Up Next" provider-name="im85288" version="1.0.0">
+    <requires>
+        <import addon="script.module.requests" version="2.9.1"/>
```
Comment (enen92) [inline service.upnext/addon.xml]: missing `xbmc.python` dependency:

```
  <requires>
    <import addon="xbmc.python" version="2.26.0"/>
  </requires>
```

File: service.upnext/resources/lib/developer.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            next_up_page.show()
+        elif utils.settings("windowMode") == "1":
+            next_up_page_simple.show()
+        elif utils.settings("windowMode") == "2":
+            still_watching_page.show()
+        elif utils.settings("windowMode") == "3":
+            still_watching_page_simple.show()
+        utils.window('service.upnext.dialog', 'true')
+
+        while (xbmc.Player().isPlaying() and not next_up_page.isCancel() and
```
Comment (enen92) [inline service.upnext/resources/lib/developer.py]: maybe avoid creating a new player instance all the time. Define `player = xbmc.Player()` above

File: service.upnext/resources/lib/utils.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
+import xbmc
+import xbmcgui
+import xbmcaddon
+import inspect
+import binascii
+import json
+
+
+def addon_id():
+    return "service.upnext"
```
Comment (enen92) [inline service.upnext/resources/lib/utils.py]: `xbmcaddon.Addon().getAddonInfo('id')`

File: service.upnext/resources/lib/utils.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+
+def addon_path():
```
Comment (enen92) [inline service.upnext/resources/lib/utils.py]: you don't need to specify the id: `xbmcaddon.Addon()` is enough. If `addon_id()` is unused after the refactoring please remove.

**Comment by enen92** [comment]
@im85288 sorry for the delay.

**Comment by enen92** [comment]
Thanks for adjusting and also for the script since I am also a user of its predecessor. Guess I have a new one to give a go :)
Cheers

**Pull request merged**